client/src/Components/NavBar/ProfileMenu.tsx

```diff
@@ -0,0 +1,61 @@
+import { createPortal } from 'react-dom';
+import { Link } from 'react-router-dom';
+import { useUser } from '../UserManagement/useUser';
+
+type Props = {
+  isOpen: boolean;
+  positionTo: SVGSVGElement | null;
+  onClose: () => void;
+};
+export function ProfileMenu({ isOpen, positionTo, onClose }: Props) {
+  const { user, handleSignOut } = useUser();
+  if (!isOpen) return null;
+
+  const r = positionTo?.getBoundingClientRect();
+  const top = r ? `${r.bottom + 4}px` : '8%';
+  const left = r ? `${r.left + r.width / 2}px` : '95%';
+
+  return createPortal(
```

Comment:
If this is the only popup you will have in your project, this is fine. But if you decide to introduce another one, you will want to extract a Popup class from here, like we did in the react-popup exercise.